SocialNetwork/SocialNetwork.API/Controllers/UserController.cs

```diff
@@ -1,51 +1,111 @@
 ﻿using Microsoft.AspNetCore.Mvc;
 using SocialNetwork.Core.Interfaces;
 using SocialNetwork.Core.Models;
-
+using System.Xml.Linq;
 namespace SocialNetwork.API.Controllers;
-
 [ApiController]
 [Route("users")]
 public class UserController : ControllerBase
 {
-    private readonly IUserService _userService;
+    private readonly IUserService _service;
 
-    public UserController(IUserService userService)
+    public UserController(IUserService service)
     {
-        _userService = userService;
+        _service = service;
     }
-
-    // localhost:port/api/users?take={take}&skip={skip}
-    [HttpGet]
-    public ActionResult<IEnumerable<User>> GetUsers([FromQuery]int skip = 0, [FromQuery]int take = 20)
+    #region Get Methods
+    [HttpGet("all")]
+    public async Task<ActionResult<IEnumerable<User>>> GetAllUsers([FromQuery] int skip = 0, [FromQuery] int take = 10)
     {
-        return Ok(_userService.GetAllUsers(skip, take));
+        try
+        {
+            return Ok( await _service.GetUsers(skip, take));
+        }
+        catch(Exception ex)
```

Comment:
If it's an Exaption then we do not know what happened. Therefore we should return status code 500 because we do not handle the error. Use badrequest and other types for defined error like ArgumentException and so on